src/js/program/background-media.ts

```diff
@@ -12,34 +18,36 @@
  * @property {Promise} nextMedia - A promise representing the media loading in the background. Resolves with the mediaController.
  * @constructor
  */
-export default function BackgroundMedia() {
-    let currentMedia = null;
-    let nextMedia = null;
+export default function BackgroundMedia(): object {
+    let currentMedia: MediaObject | null = null;
+    let nextMedia: MediaObject | null = null;
 
     return Object.defineProperties({
-        setNext(item, loadPromise) {
+        setNext(item: Item, loadPromise: Promise<any>): void {
```

Comment:
this should be `Promise<MediaController>` if i'm reading the JSDOC properly

`Resolves with the mediaController`

@robwalch confirm?

Reply:
Correct. See program-controller `backgroundLoad` which calls `setNext`.